lib/internal/fs/cp/cp.js

```diff
@@ -325,11 +325,12 @@ async function mkDirAndCopy(srcMode, src, dest, opts) {
 }
 
 async function copyDir(src, dest, opts) {
-  const dir = await readdir(src);
-  for (let i = 0; i < dir.length; i++) {
-    const item = dir[i];
-    const srcItem = join(src, item);
-    const destItem = join(dest, item);
+  const dir = await opendir(src);
+
+  for await (const dirent of dir) {
+    const { name } = dirent;
```

Comment:
```suggestion
  for await (const { name } of dir) {
```

Just a nit to save a line. Feel free to ignore.